### PR TITLE
model: remove workload labels for proxy

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR does two things
- Removes redundant "WorkloadLabels" from Proxy struct and uses "Node.Metadata.Labels" directly.
- Refactored proxy init logic in to initProxy and updateProxy in ads

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -1587,7 +1592,12 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) *EnvoyFilterWrapper {
 	// To prevent duplicate envoyfilters in case root namespace equals proxy's namespace
 	if proxy.ConfigNamespace != ps.Mesh.RootNamespace {
 		for _, efw := range ps.envoyFiltersByNamespace[proxy.ConfigNamespace] {
-			if efw.workloadSelector == nil || proxy.WorkloadLabels.IsSupersetOf(efw.workloadSelector) {
+			workloadLabels := labels.Collection{}
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context.go]: Can we remove the `Collection` as well?

Comment (ramaraochavali) [reply pilot/pkg/model/push_context.go]: ok. changed PTAL

**Comment by ramaraochavali** [comment]
@istio-wg-security-maintainers can you PTAL?

**Comment by ramaraochavali** [comment]
@istio/wg-security-maintainers can you PTAL?

File: pilot/pkg/model/context.go [hunk 2]
```diff
@@ -578,8 +568,9 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) error {
 		log.Errorf("failed to get service proxy labels: %v", err)
 		return err
 	}
-
-	node.WorkloadLabels = l
+	if len(l) > 0 {
+		node.Metadata.Labels = l[0]
```
Comment (incfly) [inline pilot/pkg/model/context.go]: it seems to be that labels.Collection is an unnecessary wrapping unless i miss anything. we always do a in place `{}` construction and also only use one from that.

just a thought, not something need to address in the PR. might something worth future clean up

Comment (ramaraochavali) [reply pilot/pkg/model/context.go]: I agree, that needs an interface change to `GetProxyWorkloadLabels` - that is why I did not do in this PR but we should clean that up. Only advantage that gives today is it provides utility functions for comparison like `SubsetOf` which can be easily implemented without that wrapper.

**Pull request merged**